offlinedatasci/main.py

```diff
@@ -204,17 +206,39 @@ def python_libraries(ods_dir):
         'pkgs': py_library_reqs,
         'python_version': '3.9.6'
     }
-    pypi_mirror.download(platform = [ 'manylinux1_x86_64'], **parameters)
-    pypi_mirror.download(platform = [ 'macosx_10_10_x86_64'], **parameters)
-    pypi_mirror.download(platform = [ 'win_amd64'], **parameters)
+    pypi_mirror.download(platform = ['manylinux1_x86_64'], **parameters)
+    pypi_mirror.download(platform = ['macosx_10_10_x86_64'], **parameters)
+    pypi_mirror.download(platform = ['win_amd64'], **parameters)
     mirror_creation_parameters = {
     'download_dir': download_dir,
     'mirror_dir': pypi_dir
     }
     pypi_mirror.create_mirror(**mirror_creation_parameters)
 
+def default_packages_python():
+    packages = {
+        "software-carpentry": ["matplotlib", "notebook", "numpy", "pandas"] ,
+        "data-science":["tensorflow", "scipy", "numpy", "pandas", "matplotlib", "keras", "scikit-learn", "pytorch", "scrapy", "beautifulsoup", "seaborn"]
+    }
+    return packages
 
+def default_packages_r():
+    packages = {
+        "data-carpentry": ["tidyverse", "RSQLite"],
+        "data-science":["dplyr", "ggplot2", "shiny", "lubridate", "knitr", "esquisse", "mlr3", "knitr", "DT"]
+    }
 
-
-
-
+    return packages
+
+def package_selection(language,custom_package_list):
+    package_language = f"default_packages_{language}"
+    language_packages_function = getattr(sys.modules[__name__], package_language)
+    language_dictionary = language_packages_function()
+    packages_to_download = []
+    for item in custom_package_list:
+        if item in [*language_dictionary]:
+            packages_to_download += language_dictionary[item]
```

Comment:
I'd probably use `packages_to_download.extend(language_dictionary[item])` since it lines up more clearly with the use of `.append` in the `else` clause.